libr/asm/arch/pyc/opcode_anal.c

```diff
@@ -0,0 +1,907 @@
+	{ "LOAD_GLOBAL", anal_LOAD_GLOBAL },
+	{ "LOAD_GLOBALS", anal_LOAD_GLOBALS },
+	{ "LOAD_LOCAL", anal_LOAD_LOCAL },
+	{ "LOAD_LOCALS", anal_LOAD_LOCALS },
+	{ "LOAD_METHOD", anal_LOAD_METHOD },
+	{ "LOAD_NAME", anal_LOAD_NAME },
+	{ "MAKE_CLOSURE", anal_MAKE_CLOSURE },
+	{ "MAKE_FUNCTION", anal_MAKE_FUNCTION },
+	{ "MAP_ADD", anal_MAP_ADD },
+	{ "NOP", anal_NOP },
+	{ "POP_BLOCK", anal_POP_BLOCK },
+	{ "POP_EXCEPT", anal_POP_EXCEPT },
+	{ "POP_FINALLY", anal_POP_FINALLY },
+	{ "POP_TOP", anal_POP_TOP },
+	{ "PRINT_EXPR", anal_PRINT_EXPR },
+	{ "PRINT_ITEM", anal_PRINT_ITEM },
+	{ "PRINT_ITEM_TO", anal_PRINT_ITEM_TO },
+	{ "PRINT_NEWLINE", anal_PRINT_NEWLINE },
+	{ "PRINT_NEWLINE_TO", anal_PRINT_NEWLINE_TO },
+	{ "RAISE_EXCEPTION", anal_RAISE_EXCEPTION },
+	{ "RAISE_VARARGS", anal_RAISE_VARARGS },
+	{ "RERAISE", anal_RERAISE },
+	{ "RESERVE_FAST", anal_RESERVE_FAST },
+	{ "RETURN_VALUE", anal_RETURN_VALUE },
+	{ "ROT_FOUR", anal_ROT_FOUR },
+	{ "ROT_THREE", anal_ROT_THREE },
+	{ "ROT_TWO", anal_ROT_TWO },
+	{ "SETUP_ANNOTATIONS", anal_SETUP_ANNOTATIONS },
+	{ "SET_ADD", anal_SET_ADD },
+	{ "SET_FUNC_ARGS", anal_SET_FUNC_ARGS },
+	{ "SET_LINENO", anal_SET_LINENO },
+	{ "SLICE_0", anal_SLICE_0 },
+	{ "SLICE_1", anal_SLICE_1 },
+	{ "SLICE_2", anal_SLICE_2 },
+	{ "SLICE_3", anal_SLICE_3 },
+	{ "STOP_CODE", anal_STOP_CODE },
+	{ "STORE_ANNOTATION", anal_STORE_ANNOTATION },
+	{ "STORE_ATTR", anal_STORE_ATTR },
+	{ "STORE_DEREF", anal_STORE_DEREF },
+	{ "STORE_FAST", anal_STORE_FAST },
+	{ "STORE_GLOBAL", anal_STORE_GLOBAL },
+	{ "STORE_LOCALS", anal_STORE_LOCALS },
+	{ "STORE_MAP", anal_STORE_MAP },
+	{ "STORE_NAME", anal_STORE_NAME },
+	{ "STORE_SLICE_0", anal_STORE_SLICE_0 },
+	{ "STORE_SLICE_1", anal_STORE_SLICE_1 },
+	{ "STORE_SLICE_2", anal_STORE_SLICE_2 },
+	{ "STORE_SLICE_3", anal_STORE_SLICE_3 },
+	{ "STORE_SUBSCR", anal_STORE_SUBSCR },
+	{ "UNARY_CALL", anal_UNARY_CALL },
+	{ "UNARY_CONVERT", anal_UNARY_CONVERT },
+	{ "UNARY_INVERT", anal_UNARY_INVERT },
+	{ "UNARY_NEGATIVE", anal_UNARY_NEGATIVE },
+	{ "UNARY_NOT", anal_UNARY_NOT },
+	{ "UNARY_POSITIVE", anal_UNARY_POSITIVE },
+	{ "UNPACK_ARG", anal_UNPACK_ARG },
+	{ "UNPACK_EX", anal_UNPACK_EX },
+	{ "UNPACK_LIST", anal_UNPACK_LIST },
+	{ "UNPACK_SEQUENCE", anal_UNPACK_SEQUENCE },
+	{ "UNPACK_TUPLE", anal_UNPACK_TUPLE },
+	{ "UNPACK_VARARG", anal_UNPACK_VARARG },
+	{ "WITH_CLEANUP", anal_WITH_CLEANUP },
+	{ "WITH_CLEANUP_FINISH", anal_WITH_CLEANUP_FINISH },
+	{ "WITH_CLEANUP_START", anal_WITH_CLEANUP_START },
+	{ "WITH_EXCEPT_START", anal_WITH_EXCEPT_START },
+	{ "YIELD_FROM", anal_YIELD_FROM },
+	{ "YIELD_VALUE", anal_YIELD_VALUE },
+	// Fix jump info
+	{ "FOR_ITER", anal_FOR_ITER },
+	{ "SETUP_LOOP", anal_SETUP_LOOP },
+	{ "SETUP_EXCEPT", anal_SETUP_EXCEPT },
+	{ "SETUP_FINALLY", anal_SETUP_FINALLY },
+	{ "SETUP_WITH", anal_SETUP_WITH },
+	{ "SETUP_ASYNC_WITH", anal_SETUP_ASYNC_WITH },
+};
+
+void anal_pyc_op (RAnalOp *op, pyc_opcode_object *op_obj, ut32 oparg) {
```

Comment:
We can. Probably fill a hashtable instead of doing that loop for every instruction

Reply:
But i understand the pr is big enough to delegate such refactorings and improvements after being merged. What do u think @ret2libc

Reply:
i think this can be an enhancement that can be addressed later

Reply:
Agree, please do this in the separate PR.

Reply:
Agree, another PR is better.